### PR TITLE
Update upgrade documentation

### PR DESCRIPTION
**Pull request by dhurley**
### Proposed changes

Update upgrade documentation based on the change in behaviour introduced by this PR https://github.com/nginx/agent/pull/352

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

**Review by craigell — APPROVED** [review]

**Review by Dean-Coakley — APPROVED** [review]

**Review by olli-holmala — APPROVED** [review]

File: hugo/content/upgrading.md [hunk 1]
```diff
@@ -10,7 +10,38 @@ categories: ["configuration"]
 doctypes: ["task"]
 ---
 
-## Upgrade the NGINX Agent
+## Upgrade the NGINX Agent from version v2.31.0 or greater
+
+{{< important >}} In version v2.31.0 of the NGINX Agent, the behavior has changed so that the NGINX Agent will automatically restart itself during an upgrade. {{< /important >}}
```
Comment (Jcahilltorre) [inline hugo/content/upgrading.md:15]:
```suggestion
{{< important >}} Starting from version v2.31.0, NGINX Agent will automatically restart itself during an upgrade. {{< /important >}}
```

Here and going forward, we don't add "the" in front of NGINX products as per the style guide.

File: hugo/content/upgrading.md [hunk 2]
```diff
@@ -10,7 +10,38 @@ categories: ["configuration"]
...
+
+To upgrade the NGINX Agent, take the following steps:
```
Comment (Jcahilltorre) [inline hugo/content/upgrading.md:17]:
```suggestion
To upgrade NGINX Agent, follow these steps:
```

File: hugo/content/upgrading.md [hunk 3]
```diff
@@ -10,7 +10,38 @@ categories: ["configuration"]
...
+
+1. Open an SSH connection to the server where you’ve installed the NGINX Agent and log in.
```
Comment (Jcahilltorre) [inline hugo/content/upgrading.md:19]:
```suggestion
1. Open an SSH connection to the server where you’ve installed NGINX Agent and log in.
```

File: hugo/content/upgrading.md [hunk 4]
```diff
@@ -10,7 +10,38 @@ categories: ["configuration"]
...
+
+1. Make a backup copy of the following locations to ensure that you can successfully recover if the upgrade has issues:
+
+    - `/etc/nginx-agent`
+    - `config_dirs` values for any configuration specified in `/etc/nginx-agent/nginx-agent.conf`
+
+1. Install the updated version of the NGINX Agent:
```
Comment (Jcahilltorre) [inline hugo/content/upgrading.md:26]:
```suggestion
1. Install the updated version of NGINX Agent:
```

File: hugo/content/upgrading.md [hunk 5]
```diff
@@ -10,7 +10,38 @@ categories: ["configuration"]
...
+        ```shell
+        sudo apt-get update
+        sudo apt-get install -y --only-upgrade nginx-agent -o Dpkg::Options::="--force-confold"
+        ```
+
+
+
+## Upgrade the NGINX Agent from a version less than v2.31.0
 
 To upgrade the NGINX Agent, take the following steps:
```
Comment (Jcahilltorre) [inline hugo/content/upgrading.md:46]:
```suggestion
## Upgrade NGINX Agent from a version less than v2.31.0

To upgrade NGINX Agent, take the following steps:
```

File: hugo/content/upgrading.md [hunk 6]
```diff
@@ -10,7 +10,38 @@ categories: ["configuration"]
 doctypes: ["task"]
 ---
 
-## Upgrade the NGINX Agent
+## Upgrade the NGINX Agent from version v2.31.0 or greater
```
Comment (Jcahilltorre) [inline hugo/content/upgrading.md:13]:
```suggestion
## Upgrade NGINX Agent from version v2.31.0 or greater
```

**Review by Jcahilltorre — APPROVED** [review]
approved with comments

**Pull request merged**